cloudtools/scripts/aws_watch_pending.py

```diff
@@ -306,7 +306,15 @@ def do_request_instance(region, moz_instance_type, price, ami, instance_config,
         groups=instance_config[region].get("security_group_ids"))
     nc = NetworkInterfaceCollection(spec)
 
-    user_data = user_data_from_template(moz_instance_type, fqdn, region)
+    if 'user_data_file' in instance_config[region]:
+        user_data = open(instance_config[region]['user_data_file']).read()
+        user_data = user_data.format(
+            hostname=name,
+            domain=instance_config[region]["domain"],
+            region_dns_atom=get_region_dns_atom(region))
+    else:
+        user_data = user_data_from_template(moz_instance_type, fqdn, region)
```

Comment:
If you refactor the function above, user_data_from_template() may need some extra love. Maybe, instead of passing a list of things to be replaced, you would pass a dict with replacements. something like
``` python
user_data_from_template({
    "moz_instance_type": moz_instance_type,
    "fqdn": fqdn, "region": region,
    "region_dns_atom": get_region_dns_atom(region)
})
```